src/Saunter/UI/AsyncApiUiMiddleware.cs

```diff
@@ -19,12 +19,20 @@ namespace Saunter.UI
 {
     public class AsyncApiUiMiddleware
     {
-        private readonly IOptions<AsyncApiOptions> _options;
+        private readonly AsyncApiOptions _options;
         private readonly StaticFileMiddleware _staticFiles;
 
-        public AsyncApiUiMiddleware(RequestDelegate next, IOptions<AsyncApiOptions> options, IWebHostEnvironment env, ILoggerFactory loggerFactory)
+
+        public AsyncApiUiMiddleware(RequestDelegate next, IOptionsSnapshot<AsyncApiOptions> options, IWebHostEnvironment env, ILoggerFactory loggerFactory, string apiName = null)
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
+            if (apiName == null)
+            {
+                _options = options.Value;
+            }
+            else
+            {
+                _options = options.Get(apiName);
```

Comment:
I think thats not possible. we could introduce a new Option class that has a list of async api options and use that one instead.

Reply:
Yes, I think this might be better to change the options class a bit to support this.
Ideally the middleware config can just be added like it is today, without specifying the documents e.g.
```cs
app.UseEndpoints(endpoints =>
{
    endpoints.MapAsyncApiDocuments();
    endpoints.MapAsyncApiUi();
});
```

I think this design is good for adding additional apis/docs:
```cs
// if only using 1 api/doc
services.AddAsyncApiSchemaGeneration(options => 
{
    options.AsyncApi = new AsyncApi();

    options.Middleware.Route = "/asyncapi/asyncapi.json";
});


// if using more than one
services.AddAsyncApiSchemaGeneration(options => 
{
    options.Middleware.Route = "asyncapi/{document}/asyncapi.json";
});
services.ConfigureNamedAsyncApi("A", options =>
{
    options.AsyncApi = new AsyncApi();
});
services.ConfigureNamedAsyncApi("B", options =>
{
    options.AsyncApi = new AsyncApi();
});
```

What do you think?

Reply:
i changed the implementation to only require one middleware. code is a bit more fiddly.

i also added support for netstandard (netcoreapp 2.x), i can remove that / prepare another pr for it I just need it for testing in my project